online_meetings/canvas_zoom_meetings.py

```diff
@@ -105,51 +105,65 @@ def get_zoom_details(self, url: str, data: Dict[str, str], course_id: int):
             logger.warn("PATTERN NOT FOUND in course, no details logged")
             logger.debug(r.text)
 
-    def zoom_course_report(self, canvas_account: int = 1, enrollment_term_id: int = 1, published: bool = True):
+    def get_zoom_course(self, course: canvasapi.course.Course) -> None:
+        # Get tabs and look for defined tool(s) that aren't hidden
+        tabs = course.get_tabs()
+        for tab in tabs:
+            # Hidden only included if true
+            if (tab.label == "Zoom" and not hasattr(tab, "hidden")):
+                logger.info("Found a course with zoom as %s", tab.id)
+
+                r = CANVAS._Canvas__requester.request("GET", _url=tab.url)
+                external_url = r.json().get("url")
+                r = requests.get(external_url)
+                # Parse out the form from the response
+                soup = bs(r.text, 'html.parser')
+                # Get the form and parse out all of the inputs
+                form = soup.find('form')
+                if not form:
+                    logger.info("Could not find a form to launch this zoom page, skipping")
+                    break
+
+                self.zoom_courses.append({'account_id': course.account_id, 'course_id': course.id, 'course_name': course.name})
+
+                fields = form.findAll('input')
+                formdata = dict((field.get('name'), field.get('value')) for field in fields)
+                # Get the URL to post back to
+                posturl = form.get('action')
+                self.get_zoom_details(posturl, formdata, course.id)
+        return None
+
+    def zoom_course_report(self, canvas_account: int = 1, enrollment_term_id: int = 0,
+                           published: bool = True, add_course_ids: list = None) -> None:
 
         account = CANVAS.get_account(canvas_account)
         # Canvas has a limit of 100 per page on this API
         per_page = 100
         # Get all published courses from the defined enrollment term
-        courses = account.get_courses(enrollment_term_id=enrollment_term_id, published=published, per_page=per_page)
-        # For testing
+        courses = []
+        if enrollment_term_id:
+            courses = account.get_courses(enrollment_term_id=enrollment_term_id, published=published, per_page=per_page)
+
+        # If there's course_id's passed in extend this to include these courses
+
```

Comment:
Maybe this comment can be moved to be with wherever you're dealing with the course IDs?

Reply:
Thanks, I originally had it going first but was thinking of adding in a process to catch dupes. I guess I can just add an extra if statement to remove dupes. Duh it's really easy.